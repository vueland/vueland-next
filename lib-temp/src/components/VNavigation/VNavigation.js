import { defineComponent, h, ref, computed } from 'vue';
import { useColors, colorProps } from '../../composable/use-colors';
import { convertToUnit } from '../../helpers';
export const VNavigation = defineComponent({
    name: 'v-navigation',
    props: {
        fixed: Boolean,
        right: Boolean,
        left: Boolean,
        onHover: Boolean,
        miniVariant: Boolean,
        expand: Boolean,
        offsetTop: {
            type: [String, Number],
            default: 0
        },
        miniVariantWidth: {
            type: [String, Number],
            default: 56
        },
        maxVariantWidth: {
            type: [String, Number],
            default: 237
        },
        ...colorProps('#ffffff')
    },
    emits: ['update:expand'],
    setup(props, { slots, emit }) {
        const { setBackgroundClassNameColor, setBackgroundCssColor } = useColors();
        const isHovered = ref(false);
        const isExpanded = computed(() => {
            if (props.onHover && isHovered.value)
                return true;
            if (props.expand)
                return true;
            if (!props.expand && props.miniVariant)
                return false;
            if (!props.expand && !props.miniVariant)
                return false;
            return !props.miniVariant;
        });
        const computedWidth = computed(() => {
            if (!isExpanded.value)
                return props.miniVariantWidth;
            return props.maxVariantWidth;
        });
        const classes = computed(() => ({
            'v-navigation': true,
            'v-navigation--expanded': isExpanded.value,
            'v-navigation--fixed': props.fixed,
            ...(props.color ? setBackgroundClassNameColor(props.color) : {})
        }));
        const styles = computed(() => ({
            width: convertToUnit(computedWidth.value),
            maxHeight: `calc(100vh - ${convertToUnit(props.offsetTop)})`,
            top: props.fixed ? convertToUnit(props.offsetTop) : '',
            left: !props.right && props.fixed ? convertToUnit(0) : '',
            right: props.right && props.fixed ? convertToUnit(0) : '',
            ...(props.color ? setBackgroundCssColor(props.color) : {})
        }));
        const genNavigationContent = () => {
            return h('div', {
                class: 'v-navigation__content'
            }, {
                default: () => slots.default?.()
            });
        };
        const onMouseenter = () => {
            if (!props.onHover || props.miniVariant)
                return;
            isHovered.value = true;
            emit('update:expand', true);
        };
        const onMouseleave = () => {
            if (!props.onHover || props.miniVariant)
                return;
            isHovered.value = false;
            emit('update:expand', false);
        };
        return () => h('aside', {
            class: classes.value,
            style: styles.value,
            onMouseenter,
            onMouseleave
        }, genNavigationContent());
    }
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVk5hdmlnYXRpb24uanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTmF2aWdhdGlvbi9WTmF2aWdhdGlvbi50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsZUFBZSxFQUFFLENBQUMsRUFBRSxHQUFHLEVBQUUsUUFBUSxFQUFTLE1BQU0sS0FBSyxDQUFBO0FBRTlELE9BQU8sRUFBRSxTQUFTLEVBQUUsVUFBVSxFQUFFLE1BQU0sNkJBQTZCLENBQUE7QUFFbkUsT0FBTyxFQUFFLGFBQWEsRUFBRSxNQUFNLGVBQWUsQ0FBQTtBQUM3QyxNQUFNLENBQUMsTUFBTSxXQUFXLEdBQUcsZUFBZSxDQUFDO0lBQ3pDLElBQUksRUFBRSxjQUFjO0lBQ3BCLEtBQUssRUFBRTtRQUNMLEtBQUssRUFBRSxPQUFPO1FBQ2QsS0FBSyxFQUFFLE9BQU87UUFDZCxJQUFJLEVBQUUsT0FBTztRQUNiLE9BQU8sRUFBRSxPQUFPO1FBQ2hCLFdBQVcsRUFBRSxPQUFPO1FBQ3BCLE1BQU0sRUFBRSxPQUFPO1FBQ2YsU0FBUyxFQUFFO1lBQ1QsSUFBSSxFQUFFLENBQUUsTUFBTSxFQUFFLE1BQU0sQ0FBRTtZQUN4QixPQUFPLEVBQUUsQ0FBQztTQUNYO1FBQ0QsZ0JBQWdCLEVBQUU7WUFDaEIsSUFBSSxFQUFFLENBQUUsTUFBTSxFQUFFLE1BQU0sQ0FBRTtZQUN4QixPQUFPLEVBQUUsRUFBRTtTQUNaO1FBQ0QsZUFBZSxFQUFFO1lBQ2YsSUFBSSxFQUFFLENBQUUsTUFBTSxFQUFFLE1BQU0sQ0FBRTtZQUN4QixPQUFPLEVBQUUsR0FBRztTQUNiO1FBQ0QsR0FBRyxVQUFVLENBQUMsU0FBUyxDQUFDO0tBQ3pCO0lBQ0QsS0FBSyxFQUFFLENBQUUsZUFBZSxDQUFFO0lBRTFCLEtBQUssQ0FBQyxLQUFLLEVBQUUsRUFBRSxLQUFLLEVBQUUsSUFBSSxFQUFFO1FBQzFCLE1BQU0sRUFBRSwyQkFBMkIsRUFBRSxxQkFBcUIsRUFBRSxHQUFHLFNBQVMsRUFBRSxDQUFBO1FBQzFFLE1BQU0sU0FBUyxHQUFHLEdBQUcsQ0FBVSxLQUFLLENBQUMsQ0FBQTtRQUVyQyxNQUFNLFVBQVUsR0FBRyxRQUFRLENBQUMsR0FBRyxFQUFFO1lBQy9CLElBQUksS0FBSyxDQUFDLE9BQU8sSUFBSSxTQUFTLENBQUMsS0FBSztnQkFBRSxPQUFPLElBQUksQ0FBQTtZQUNqRCxJQUFJLEtBQUssQ0FBQyxNQUFNO2dCQUFFLE9BQU8sSUFBSSxDQUFBO1lBQzdCLElBQUksQ0FBQyxLQUFLLENBQUMsTUFBTSxJQUFJLEtBQUssQ0FBQyxXQUFXO2dCQUFFLE9BQU8sS0FBSyxDQUFBO1lBQ3BELElBQUksQ0FBQyxLQUFLLENBQUMsTUFBTSxJQUFJLENBQUMsS0FBSyxDQUFDLFdBQVc7Z0JBQUUsT0FBTyxLQUFLLENBQUE7WUFDckQsT0FBTyxDQUFDLEtBQUssQ0FBQyxXQUFXLENBQUE7UUFDM0IsQ0FBQyxDQUFDLENBQUE7UUFFRixNQUFNLGFBQWEsR0FBRyxRQUFRLENBQWtCLEdBQUcsRUFBRTtZQUNuRCxJQUFJLENBQUMsVUFBVSxDQUFDLEtBQUs7Z0JBQUUsT0FBTyxLQUFLLENBQUMsZ0JBQWdCLENBQUE7WUFDcEQsT0FBTyxLQUFLLENBQUMsZUFBZSxDQUFBO1FBQzlCLENBQUMsQ0FBQyxDQUFBO1FBRUYsTUFBTSxPQUFPLEdBQUcsUUFBUSxDQUEwQixHQUFHLEVBQUUsQ0FBQyxDQUFDO1lBQ3ZELGNBQWMsRUFBRSxJQUFJO1lBQ3BCLHdCQUF3QixFQUFFLFVBQVUsQ0FBQyxLQUFLO1lBQzFDLHFCQUFxQixFQUFFLEtBQUssQ0FBQyxLQUFLO1lBQ2xDLEdBQUcsQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLENBQUMsQ0FBQywyQkFBMkIsQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQztTQUNqRSxDQUFDLENBQUMsQ0FBQTtRQUVILE1BQU0sTUFBTSxHQUFHLFFBQVEsQ0FBeUIsR0FBRyxFQUFFLENBQUMsQ0FBQztZQUNyRCxLQUFLLEVBQUUsYUFBYSxDQUFDLGFBQWEsQ0FBQyxLQUFLLENBQUM7WUFDekMsU0FBUyxFQUFFLGdCQUFpQixhQUFhLENBQUMsS0FBSyxDQUFDLFNBQVMsQ0FBRSxHQUFHO1lBQzlELEdBQUcsRUFBRSxLQUFLLENBQUMsS0FBSyxDQUFDLENBQUMsQ0FBQyxhQUFhLENBQUMsS0FBSyxDQUFDLFNBQVMsQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFO1lBQ3RELElBQUksRUFBRSxDQUFDLEtBQUssQ0FBQyxLQUFLLElBQUksS0FBSyxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsYUFBYSxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFO1lBQ3pELEtBQUssRUFBRSxLQUFLLENBQUMsS0FBSyxJQUFJLEtBQUssQ0FBQyxLQUFLLENBQUMsQ0FBQyxDQUFDLGFBQWEsQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUMsRUFBRTtZQUN6RCxHQUFHLENBQUMsS0FBSyxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMscUJBQXFCLENBQUMsS0FBSyxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFLENBQUM7U0FDM0QsQ0FBQyxDQUFDLENBQUE7UUFFSCxNQUFNLG9CQUFvQixHQUFHLEdBQVUsRUFBRTtZQUN2QyxPQUFPLENBQUMsQ0FBQyxLQUFLLEVBQUU7Z0JBQ2QsS0FBSyxFQUFFLHVCQUF1QjthQUMvQixFQUFFO2dCQUNELE9BQU8sRUFBRSxHQUFHLEVBQUUsQ0FBQyxLQUFLLENBQUMsT0FBTyxFQUFFLEVBQUU7YUFDakMsQ0FBQyxDQUFBO1FBQ0osQ0FBQyxDQUFBO1FBRUQsTUFBTSxZQUFZLEdBQUcsR0FBRyxFQUFFO1lBQ3hCLElBQUksQ0FBQyxLQUFLLENBQUMsT0FBTyxJQUFJLEtBQUssQ0FBQyxXQUFXO2dCQUFFLE9BQU07WUFDL0MsU0FBUyxDQUFDLEtBQUssR0FBRyxJQUFJLENBQUE7WUFDdEIsSUFBSSxDQUFDLGVBQWUsRUFBRSxJQUFJLENBQUMsQ0FBQTtRQUM3QixDQUFDLENBQUE7UUFFRCxNQUFNLFlBQVksR0FBRyxHQUFHLEVBQUU7WUFDeEIsSUFBSSxDQUFDLEtBQUssQ0FBQyxPQUFPLElBQUksS0FBSyxDQUFDLFdBQVc7Z0JBQUUsT0FBTTtZQUMvQyxTQUFTLENBQUMsS0FBSyxHQUFHLEtBQUssQ0FBQTtZQUN2QixJQUFJLENBQUMsZUFBZSxFQUFFLEtBQUssQ0FBQyxDQUFBO1FBQzlCLENBQUMsQ0FBQTtRQUVELE9BQU8sR0FBRyxFQUFFLENBQUMsQ0FBQyxDQUFDLE9BQU8sRUFBRTtZQUN0QixLQUFLLEVBQUUsT0FBTyxDQUFDLEtBQUs7WUFDcEIsS0FBSyxFQUFFLE1BQU0sQ0FBQyxLQUFLO1lBQ25CLFlBQVk7WUFDWixZQUFZO1NBQ2IsRUFBRSxvQkFBb0IsRUFBRSxDQUFDLENBQUE7SUFDNUIsQ0FBQztDQUNGLENBQUMsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IGRlZmluZUNvbXBvbmVudCwgaCwgcmVmLCBjb21wdXRlZCwgVk5vZGUgfSBmcm9tICd2dWUnXG4vLyBDb21wb3NhYmxlc1xuaW1wb3J0IHsgdXNlQ29sb3JzLCBjb2xvclByb3BzIH0gZnJvbSAnLi4vLi4vY29tcG9zYWJsZS91c2UtY29sb3JzJ1xuLy8gSGVscGVyc1xuaW1wb3J0IHsgY29udmVydFRvVW5pdCB9IGZyb20gJy4uLy4uL2hlbHBlcnMnXG5leHBvcnQgY29uc3QgVk5hdmlnYXRpb24gPSBkZWZpbmVDb21wb25lbnQoe1xuICBuYW1lOiAndi1uYXZpZ2F0aW9uJyxcbiAgcHJvcHM6IHtcbiAgICBmaXhlZDogQm9vbGVhbixcbiAgICByaWdodDogQm9vbGVhbixcbiAgICBsZWZ0OiBCb29sZWFuLFxuICAgIG9uSG92ZXI6IEJvb2xlYW4sXG4gICAgbWluaVZhcmlhbnQ6IEJvb2xlYW4sXG4gICAgZXhwYW5kOiBCb29sZWFuLFxuICAgIG9mZnNldFRvcDoge1xuICAgICAgdHlwZTogWyBTdHJpbmcsIE51bWJlciBdLFxuICAgICAgZGVmYXVsdDogMFxuICAgIH0sXG4gICAgbWluaVZhcmlhbnRXaWR0aDoge1xuICAgICAgdHlwZTogWyBTdHJpbmcsIE51bWJlciBdLFxuICAgICAgZGVmYXVsdDogNTZcbiAgICB9LFxuICAgIG1heFZhcmlhbnRXaWR0aDoge1xuICAgICAgdHlwZTogWyBTdHJpbmcsIE51bWJlciBdLFxuICAgICAgZGVmYXVsdDogMjM3XG4gICAgfSxcbiAgICAuLi5jb2xvclByb3BzKCcjZmZmZmZmJylcbiAgfSxcbiAgZW1pdHM6IFsgJ3VwZGF0ZTpleHBhbmQnIF0sXG5cbiAgc2V0dXAocHJvcHMsIHsgc2xvdHMsIGVtaXQgfSk6ICgpID0+IFZOb2RlIHtcbiAgICBjb25zdCB7IHNldEJhY2tncm91bmRDbGFzc05hbWVDb2xvciwgc2V0QmFja2dyb3VuZENzc0NvbG9yIH0gPSB1c2VDb2xvcnMoKVxuICAgIGNvbnN0IGlzSG92ZXJlZCA9IHJlZjxib29sZWFuPihmYWxzZSlcblxuICAgIGNvbnN0IGlzRXhwYW5kZWQgPSBjb21wdXRlZCgoKSA9PiB7XG4gICAgICBpZiAocHJvcHMub25Ib3ZlciAmJiBpc0hvdmVyZWQudmFsdWUpIHJldHVybiB0cnVlXG4gICAgICBpZiAocHJvcHMuZXhwYW5kKSByZXR1cm4gdHJ1ZVxuICAgICAgaWYgKCFwcm9wcy5leHBhbmQgJiYgcHJvcHMubWluaVZhcmlhbnQpIHJldHVybiBmYWxzZVxuICAgICAgaWYgKCFwcm9wcy5leHBhbmQgJiYgIXByb3BzLm1pbmlWYXJpYW50KSByZXR1cm4gZmFsc2VcbiAgICAgIHJldHVybiAhcHJvcHMubWluaVZhcmlhbnRcbiAgICB9KVxuXG4gICAgY29uc3QgY29tcHV0ZWRXaWR0aCA9IGNvbXB1dGVkPG51bWJlciB8IHN0cmluZz4oKCkgPT4ge1xuICAgICAgaWYgKCFpc0V4cGFuZGVkLnZhbHVlKSByZXR1cm4gcHJvcHMubWluaVZhcmlhbnRXaWR0aFxuICAgICAgcmV0dXJuIHByb3BzLm1heFZhcmlhbnRXaWR0aFxuICAgIH0pXG5cbiAgICBjb25zdCBjbGFzc2VzID0gY29tcHV0ZWQ8UmVjb3JkPHN0cmluZywgYm9vbGVhbj4+KCgpID0+ICh7XG4gICAgICAndi1uYXZpZ2F0aW9uJzogdHJ1ZSxcbiAgICAgICd2LW5hdmlnYXRpb24tLWV4cGFuZGVkJzogaXNFeHBhbmRlZC52YWx1ZSxcbiAgICAgICd2LW5hdmlnYXRpb24tLWZpeGVkJzogcHJvcHMuZml4ZWQsXG4gICAgICAuLi4ocHJvcHMuY29sb3IgPyBzZXRCYWNrZ3JvdW5kQ2xhc3NOYW1lQ29sb3IocHJvcHMuY29sb3IpIDoge30pXG4gICAgfSkpXG5cbiAgICBjb25zdCBzdHlsZXMgPSBjb21wdXRlZDxSZWNvcmQ8c3RyaW5nLCBzdHJpbmc+PigoKSA9PiAoe1xuICAgICAgd2lkdGg6IGNvbnZlcnRUb1VuaXQoY29tcHV0ZWRXaWR0aC52YWx1ZSksXG4gICAgICBtYXhIZWlnaHQ6IGBjYWxjKDEwMHZoIC0gJHsgY29udmVydFRvVW5pdChwcm9wcy5vZmZzZXRUb3ApIH0pYCxcbiAgICAgIHRvcDogcHJvcHMuZml4ZWQgPyBjb252ZXJ0VG9Vbml0KHByb3BzLm9mZnNldFRvcCkgOiAnJyxcbiAgICAgIGxlZnQ6ICFwcm9wcy5yaWdodCAmJiBwcm9wcy5maXhlZCA/IGNvbnZlcnRUb1VuaXQoMCkgOiAnJyxcbiAgICAgIHJpZ2h0OiBwcm9wcy5yaWdodCAmJiBwcm9wcy5maXhlZCA/IGNvbnZlcnRUb1VuaXQoMCkgOiAnJyxcbiAgICAgIC4uLihwcm9wcy5jb2xvciA/IHNldEJhY2tncm91bmRDc3NDb2xvcihwcm9wcy5jb2xvcikgOiB7fSlcbiAgICB9KSlcblxuICAgIGNvbnN0IGdlbk5hdmlnYXRpb25Db250ZW50ID0gKCk6IFZOb2RlID0+IHtcbiAgICAgIHJldHVybiBoKCdkaXYnLCB7XG4gICAgICAgIGNsYXNzOiAndi1uYXZpZ2F0aW9uX19jb250ZW50J1xuICAgICAgfSwge1xuICAgICAgICBkZWZhdWx0OiAoKSA9PiBzbG90cy5kZWZhdWx0Py4oKVxuICAgICAgfSlcbiAgICB9XG5cbiAgICBjb25zdCBvbk1vdXNlZW50ZXIgPSAoKSA9PiB7XG4gICAgICBpZiAoIXByb3BzLm9uSG92ZXIgfHwgcHJvcHMubWluaVZhcmlhbnQpIHJldHVyblxuICAgICAgaXNIb3ZlcmVkLnZhbHVlID0gdHJ1ZVxuICAgICAgZW1pdCgndXBkYXRlOmV4cGFuZCcsIHRydWUpXG4gICAgfVxuXG4gICAgY29uc3Qgb25Nb3VzZWxlYXZlID0gKCkgPT4ge1xuICAgICAgaWYgKCFwcm9wcy5vbkhvdmVyIHx8IHByb3BzLm1pbmlWYXJpYW50KSByZXR1cm5cbiAgICAgIGlzSG92ZXJlZC52YWx1ZSA9IGZhbHNlXG4gICAgICBlbWl0KCd1cGRhdGU6ZXhwYW5kJywgZmFsc2UpXG4gICAgfVxuXG4gICAgcmV0dXJuICgpID0+IGgoJ2FzaWRlJywge1xuICAgICAgY2xhc3M6IGNsYXNzZXMudmFsdWUsXG4gICAgICBzdHlsZTogc3R5bGVzLnZhbHVlLFxuICAgICAgb25Nb3VzZWVudGVyLFxuICAgICAgb25Nb3VzZWxlYXZlXG4gICAgfSwgZ2VuTmF2aWdhdGlvbkNvbnRlbnQoKSlcbiAgfVxufSlcbiJdfQ==