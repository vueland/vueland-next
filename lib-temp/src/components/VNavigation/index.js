export { VNavigation } from './VNavigation';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTmF2aWdhdGlvbi9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsV0FBVyxFQUFFLE1BQU0sZUFBZSxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0IHsgVk5hdmlnYXRpb24gfSBmcm9tICcuL1ZOYXZpZ2F0aW9uJ1xuIl19