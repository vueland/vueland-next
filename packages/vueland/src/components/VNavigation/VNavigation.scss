@import "../../styles/scss/modules/global";

.v-navigation {
  display: flex;
  flex-direction: column;
  height: 100vh;
  overflow-y: auto;
  overflow-x: hidden;
  max-width: 100%;
  transition: .3s width $primary-transition;

  &--base {
    background-color: var(--base);
  }

  &--fixed {
    position: fixed;
  }

  &--absolute {
    position: absolute;
  }
}
