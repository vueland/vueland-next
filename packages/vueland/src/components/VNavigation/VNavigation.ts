import { defineComponent, h, ref, computed, VNode } from 'vue'
// Composables
import { useColors, colorProps } from '../../composables/use-colors'
// Helpers
import { convertToUnit } from '../../helpers'

export default defineComponent({
  name: 'v-navigation',
  props: {
    fixed: Boolean,
    right: Boolean,
    left: Boolean,
    onHover: Boolean,
    miniVariant: Boolean,
    expand: Boolean,
    offsetTop: {
      type: [String, Number],
      default: 0,
    },
    miniVariantWidth: {
      type: [String, Number],
      default: 56,
    },
    maxVariantWidth: {
      type: [String, Number],
      default: 237,
    },
    ...colorProps(),
  },
  emits: ['update:expand'],

  setup(props, { slots, emit }): () => VNode {
    const { setBackgroundClassNameColor, setBackgroundCssColor } = useColors()
    const isHovered = ref<boolean>(false)

    const isExpanded = computed(() => {
      if (props.onHover && isHovered.value) {
        return true
      }

      if (props.expand) {
        return true
      }

      if (!props.expand && props.miniVariant) {
        return false
      }

      if (!props.expand && !props.miniVariant) {
        return false
      }

      return !props.miniVariant
    })

    const computedWidth = computed<number | string>(() => {
      if (!isExpanded.value) return props.miniVariantWidth
      return props.maxVariantWidth
    })

    const classes = computed<Record<string, boolean>>(() => ({
      'v-navigation': true,
      'v-navigation--expanded': isExpanded.value,
      'v-navigation--fixed': props.fixed,
      'v-navigation--base': !props.color,
      ...(props.color ? setBackgroundClassNameColor(props.color) : {}),
    }))

    const styles = computed<Record<string, string>>(() => ({
      width: convertToUnit(computedWidth.value),
      maxHeight: `calc(100vh - ${ convertToUnit(props.offsetTop) })`,
      top: props.fixed ? convertToUnit(props.offsetTop) : '',
      left: !props.right && props.fixed ? convertToUnit(0) : '',
      right: props.right && props.fixed ? convertToUnit(0) : '',
      ...(props.color ? setBackgroundCssColor(props.color) : {}),
    }))

    const genNavigationContent = (): VNode => {
      return h('div', {
        class: 'v-navigation__content',
      }, {
        default: () => slots.default?.(),
      })
    }

    const onMouseenter = () => {
      if (!props.onHover || props.miniVariant) {
        return
      }

      isHovered.value = true
      emit('update:expand', true)
    }

    const onMouseleave = () => {
      if (!props.onHover || props.miniVariant) {
        return
      }

      isHovered.value = false
      emit('update:expand', false)
    }

    return () => h('aside', {
      class: classes.value,
      style: styles.value,
      onMouseenter,
      onMouseleave,
    }, genNavigationContent())
  },
})
