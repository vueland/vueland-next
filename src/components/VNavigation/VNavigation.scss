@import "../../styles/scss/modules/global";

.v-navigation {
  display: flex;
  flex-direction: column;
  height: 100vh;
  overflow-y: auto;
  overflow-x: hidden;
  max-width: 100%;
  transition: .3s width $primary-transition;
  z-index: 7;
  background-color: var(--base);

  &--fixed {
    position: fixed;
  }
}
