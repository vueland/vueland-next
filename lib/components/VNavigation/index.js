import VNavigation from './VNavigation';
export { VNavigation };
export default VNavigation;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi8uLi8uLi9wYWNrYWdlcy92dWVsYW5kL3NyYy9jb21wb25lbnRzL1ZOYXZpZ2F0aW9uL2luZGV4LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sV0FBVyxNQUFNLGVBQWUsQ0FBQTtBQUV2QyxPQUFPLEVBQUUsV0FBVyxFQUFFLENBQUE7QUFDdEIsZUFBZSxXQUFXLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgVk5hdmlnYXRpb24gZnJvbSAnLi9WTmF2aWdhdGlvbidcblxuZXhwb3J0IHsgVk5hdmlnYXRpb24gfVxuZXhwb3J0IGRlZmF1bHQgVk5hdmlnYXRpb25cbiJdfQ==