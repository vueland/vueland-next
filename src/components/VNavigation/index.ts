import VNavigation from './VNavigation'

export { VNavigation }
export default VNavigation
