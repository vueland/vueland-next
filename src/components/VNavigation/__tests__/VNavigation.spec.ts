import { mount } from '@vue/test-utils'
import { VNavigation } from '../VNavigation'
import 'regenerator-runtime'

describe('VNavigation', () => {
  const mountFunction = (options = {}) => mount(VNavigation, {
    ...options
  })

  it('should mount component and match snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.element.classList).toContain('v-navigation')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should test expanded state with default sizes', async () => {
    const wrapper = mountFunction({ props: { expand: true } })

    expect((wrapper.element as HTMLElement).style.width).toEqual('237px')
    expect(wrapper.element.classList).toContain('v-navigation--expanded')

    await wrapper.setProps({ expand: false })
    expect((wrapper.element as HTMLElement).style.width).toEqual('56px')
  })

  it('should test expanded state with custom sizes', async () => {
    const wrapper = mountFunction({
      props: { miniVariantWidth: 60, maxVariantWidth: 260, expand: true }
    })

    expect((wrapper.element as HTMLElement).style.width).toEqual('260px')
    expect(wrapper.element.classList).toContain('v-navigation--expanded')

    await wrapper.setProps({ expand: false })
    expect((wrapper.element as HTMLElement).style.width).toEqual('60px')
  })

  it('should test on hover', async () => {
    const wrapper = mountFunction({
      props: { onHover: true }
    })

    expect((wrapper.element as HTMLElement).style.width).toEqual('56px')

    await wrapper.trigger('mouseenter')

    expect((wrapper.element as HTMLElement).style.width).toEqual('237px')
  })

  it('should test mini variant', async () => {
    const wrapper = mountFunction({
      props: { onHover: true, miniVariant: true }
    })

    expect((wrapper.element as HTMLElement).style.width).toEqual('56px')

    await wrapper.trigger('mouseenter')

    expect((wrapper.element as HTMLElement).style.width).toEqual('56px')
  })

  it('should test color prop', () => {
    const wrapper = mountFunction({
      props: { color: 'red darken-1' }
    })

    expect(wrapper.element.className).toContain('red darken-1')
  })

  it('should test fixed positions', async () => {
    const wrapper = mountFunction({
      props: { fixed: true }
    })

    expect(wrapper.element.className).toContain('v-navigation--fixed')

    await wrapper.setProps({ left: true })
    expect(wrapper.find('.v-navigation').attributes('style')).toContain('left: 0px;')

    await wrapper.setProps({ right: true, left: false })
    expect(wrapper.find('.v-navigation').attributes('style')).toContain('right: 0px;')
  })

  it('should test offset-top in fixed position', async () => {
    const wrapper = mountFunction({
      props: { offsetTop: 75, fixed: true }
    })

    expect(wrapper.find('.v-navigation').attributes('style')).toContain('top: 75px;')
    expect(wrapper.find('.v-navigation').attributes('style')).toContain('height: calc(100vh - 75px);')
  })
})
