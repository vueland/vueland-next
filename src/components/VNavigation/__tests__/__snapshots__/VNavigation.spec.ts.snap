// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VNavigation should mount component and match snapshot 1`] = `
<aside class="v-navigation"
       style="width: 56px; max-height: calc(100vh - 0px); background-color: rgb(255, 255, 255); border-color: #ffffff;"
>
  <div class="v-navigation__content">
  </div>
</aside>
`;
